rusoto/credential/src/instance_metadata.rs

```diff
@@ -1,73 +1,108 @@
 //! The Credentials Provider for an AWS Resource's IAM Role.
 
-use hyper::Uri;
-use std::time::Duration as StdDuration;
-use tokio_core::reactor::Core;
+use futures::{Future, Poll};
+use futures::future::{FutureResult, result};
+use hyper::{Client, Uri};
+use hyper::client::HttpConnector;
+use tokio_core::reactor::Handle;
 
-use {AwsCredentials, CredentialsError, ProvideAwsCredentials, ProvideTimeoutableAwsCredentials,
+use {AwsCredentials, CredentialsError, ProvideAwsCredentials,
```

Comment:
So I'm all for redoing my initial async credentials crate work (because honestly it was pretty clunky, and just to see a very basic implementation). But the way I'm reading this we're actually _*losing*_ timeouts with credentials? If so I know we don't want to do that, as several people I work with (as well as I'm sure others) have already built ontop of this functionality since it's now released.

Reply:
Apologies, I didn't mean to imply that I want to get rid of the timeout mechanism, I was merely trying to get the basic async structure right first.

I'm open to either initiate the discussion around how to add timeouts as part of this PR, or to commit to implementing timeouts in a second PR before the next rusoto release under your guidance.

Let me know which one you'd prefer.